models/model.py

```diff
@@ -100,89 +124,76 @@ def get_lr_factor(self, epoch):
-        super().__init__()
-
-        model = model_hparams['model']
-        if model == 'roberta':
-            self.encoder = TransformerModel()
-        else:
-            raise ValueError("Transformer type '%s' is not supported." % model)
-
-        cf_hidden_dim = model_hparams['cf_hid_dim']
-
-        self.classifier = nn.Sequential(
-            # nn.Dropout(model_hparams['dropout']),     # TODO: maybe add later
-            nn.Linear(self.encoder.hidden_size, cf_hidden_dim),
-            nn.ReLU(),
-            nn.Linear(cf_hidden_dim, model_hparams['num_classes'])
-        )
-
-    # needs to be there to catch additional parameter modeL
-    # noinspection PyUnusedLocal
-    def forward(self, batch, **kwargs):
-        inputs, attention_mask = batch['input_ids'], batch['attention_mask']
-
-        out = self.encoder(inputs, attention_mask)
-        out = self.classifier(out)
-        return out, batch['labels']
-
-
-class TransformerModel(nn.Module):
-
+class RobertaEncoder(nn.Module):
     def __init__(self):
         super().__init__()
 
         # transformer_config = model_hparams['transformer_config']
         self.model = RobertaModel.from_pretrained('roberta-base')
 
-        # this is fixed for all base models
-        self.hidden_size = 768
-
         # this model is in eval per default, we want to fine-tune it but only the top layers
         self.model.train()
 
         # only freezing the encoder parameters / weights of the head layers
         for param in self.model.base_model.parameters():
             param.requires_grad = False
 
-    def forward(self, inputs, attention_mask=None):
+    def forward(self, batch):
+        inputs, attention_mask = batch['input_ids'], batch['attention_mask']
         # returns a tuple of torch.FloatTensor comprising various elements depending on the (RobertaConfig) and inputs.
         hidden_states = self.model(inputs, attention_mask)
 
         # b_size x hid_size
-        cls_token_state = hidden_states[1]
-
-        return cls_token_state
-
+        out = hidden_states[1]
+        return out, batch['labels']
 
-class GraphNet(torch.nn.Module):
-    def __init__(self, num_nodes):
-        super(GraphNet, self).__init__()
-        self.conv1 = GCNConv(num_nodes, 200)
-        self.conv2 = GCNConv(200, 8)
+class PureGraphEncoder(nn.Module):
+    def __init__(self, input_dim, hidden_dim):
+        super(PureGraphEncoder, self).__init__()
+        self.conv1 = GCNConv(input_dim, hidden_dim)
+        self.conv2 = GCNConv(hidden_dim, hidden_dim)
 
-    def forward(self, data):
+    def forward(self, data, mode):
         x, edge_index, edge_weight = data.x, data.edge_index, data.edge_attr
+        edge_index, edge_weight = data.edge_index, data.edge_attr
+        doc_feats = data.doc_features
+        word_feats = data.word_features
+        word_feats = self.linlay(word_feats)
+        x = torch.cat((doc_feats, word_feats))
         x = self.conv1(x, edge_index, edge_weight)
```

Comment:
So the actual second line of this function is this, everything in between should be deleted